doc/sphinx/proof-engine/tactics.rst

```diff
@@ -51,6 +51,9 @@ specified, the default selector is used.
      tactic_invocation : `toplevel_selector` : `tactic`.
                        : `tactic`.
 
+.. todo: selectors on commands, such as
+   Check, Search, SearchHead, SearchPattern, SearchRewrite.
```

Comment:
According to `fullGrammar`, selectors are supported on `query_command`:

```
tactic_mode: [
| OPT toplevel_selector G_vernac.query_command
| OPT toplevel_selector "{"
| OPT ltac_selector OPT ltac_info tactic ltac_use_default
| "par" ":" OPT ltac_info tactic ltac_use_default
]

ltac_selector: [
| toplevel_selector
]
```

These `query_command` include `Eval`, `Compute` and `About` in addition to the ones you listed:

```
query_command: [
| "Eval" red_expr "in" lconstr "."
| "Compute" lconstr "."
| "Check" lconstr "."
| "About" smart_global OPT univ_name_list "."
| "SearchHead" constr_pattern in_or_out_modules "."
| "SearchPattern" constr_pattern in_or_out_modules "."
| "SearchRewrite" constr_pattern in_or_out_modules "."
| "Search" searchabout_query searchabout_queries "."
]
```

They are merged into commands in `common.edit_mlg` but it would be great to keep some trace that of their special status. Currently, the commands from this list are mixed in the documentation with other query commands like `Locate`, `Print` and `Inspect`. I think it would make sense to better distinguish between the two categories because the first category can also be applied to an object which is locally defined in an interactive proof, whereas the second category can only be applied to globally defined objects (thus the difference in support or not of the goal selector).

Reply:
> it would be great to keep some trace that of their special status
> it would make sense to better distinguish between the two categories

Which would be comments in the text, not unlike what we did for attributes, right?  (Or explicitly showing the selector in the syntax for each of these.)  Eventually we may want to capture these details in metadata.

> Currently, the commands from this list are mixed in the documentation with other query commands like `Locate`, `Print` and `Inspect`.

Are you suggesting putting all these commands together in a "Query commands" section?  If so, where would you put it?

Reply:
> Which would be comments in the text

I rather meant some annotations like what you tried to do to distinguish tactics and tactical, but simply putting all these commands together in a section would be enough.

>Are you suggesting putting all these commands together in a "Query commands" section? If so, where would you put it?

Yes, or more precisely:
- create a "Query commands" section
- separate it into two subsections, one for queries that work with local objects in proof mode, and one for the rest
- it doesn't matter where exactly it is put because the next step will be to split this chapter into several pages, and this will become a standalone page.

Reply:
> separate it into two subsections, one for queries that work with local objects in proof mode, and one for the rest

Maybe just tell me what goes where.  Not planning to update the wording in this PR (much to update!).

Not clear what category About and Compute falls, since their descriptions doesn't mention anything about goals that could be selected.

Reply:
> Not clear what category About and Compute fall into, since their descriptions doesn't mention anything about goals that could be selected.

But goal selectors matter to them as well:

```
Goal (forall x, x = 0) /\ (forall x, x = true).
split; intro.
About x.
(*
x:nat

Hypothesis of the goal context.
*)
2: About x.
(*
x:bool

Hypothesis of the goal context.
*)
Compute (0 + x).
(*
     = x
     : nat
*)
Fail 2: Compute (0 + x).
(*
The command has indeed failed with message:
In environment
x : bool
The term "x" has type "bool" while it is expected to have type "nat".
*)
```

>Or do you mean just move all 8 query commands together?

Yes, let's just have a single section that gathers these 8 commands together, so that the explanation on goal selectors can be given once and for all at the top.

Reply:
I tweaked the wording for About and Compute to mention the current goal (if a proof is open).  Perhaps eventually we should have a brief central explanation of what objects are accessible while a proof is open.

I'll will move these together